### ISSUE TITLE
Feature Request: support Android Instant Apps via amp-app-banner

### ISSUE THREAD

**Issue by sebastianbenz** [ampproject/amphtml#18997]
We should make it possible to launch an [Android Instant Apps](https://developer.android.com/topic/google-play-instant/) via amp-app-banner. This would make it possible for publishers to meaningfully combine their AMP with their Instant App experience. The current amp-app-banner implementation displays the app in the Play Store, but won't launch an instant app if one is available. 

### Proposal

Add a new attribute `instant-app` to the `amp-app-banner` extension. When this flag is set, the open-button will directly launch the instant app when it's clicked. 

### Implementation

Implementation should be very straightforward as the Play Store supports a [new URL scheme](https://developer.android.com/distribute/marketing-tools/linking-to-google-play#Instant) for launching an instant app from the browser. It works by appending `&launch=true` to the regular play store url.

Example: https://play.google.com/store/apps/details?id=com.yodo1.rodeo.safari&launch=true

We simply need to change the way we create the intent URL for Android:

```
let intentUrl;
if (this.isInstantApp) {
  intentUrl = `https://play.google.com/store/apps/details?id=${appId}?launch=true&path=${pathname}`;  
} else {
  intentUrl = `android-app://${appId}/${cleanProtocol}/${host}${pathname}`;
}
return intentUrl;
```
https://github.com/ampproject/amphtml/blob/master/extensions/amp-app-banner/0.1/amp-app-banner.js#L460

There is one caveat: the play store app doesn't (yet) follow deep links and app developers need to handle the `&path=${pathname}` by themselves.

### Example

Here is a [sample page](https://amp-instant-app.glitch.me/simple.html) emulating the behavior using amp-iframe. The `View in App` button will launch an instant app (if enabled on Android).

//cc @ericlindley-g

**Comment by aghassemi** [comment]
@sebastianbenz thanks for providing the details. A lot of things we are trying to do with `amp-app-banner` is already supported  by https://firebase.google.com/docs/dynamic-links/ I wonder if we should encourage folks to use that with a simple `<a>`.

**Comment by aghassemi** [comment]
deduping with https://github.com/ampproject/amphtml/issues/6411
